### PR TITLE
Sort keys in the reverse order of `code unit less than`

### LINKED ISSUE #181: longer-or-code-unit-less-than is not a total order
The order used to sort scope keys etc., i.e.
https://wicg.github.io/import-maps/#longer-or-code-unit-less-than
> if a’s length is greater than b’s length, or if a is code unit less than b.

is not a total order. For example,

- `"A"` is longer-or-code-unit-less-than `"AAA"`
- `"AAA"` is longer-or-code-unit-less-than `"A"`

Probably this should be something like

> if a’s length is greater than b’s length, or if a’s length is equal to b’s length and a is code unit less than b.

or simply

> if b is code unit less than a.

Probably the latter one is better, because it's simple to understand and implement.
I'll create a PR shortly.

(Memo: `code unit less than` also seems to have a spec bug)

### PR DESCRIPTION
**Pull request by hiroshige-g**
Fixes #181.

***
<a href="https://pr-preview.s3.amazonaws.com/hiroshige-g/import-maps/pull/182.html" title="Last updated on Oct 15, 2019, 6:47 AM UTC (cb51eb7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/import-maps/182/eacf4ea...hiroshige-g:cb51eb7.html" title="Last updated on Oct 15, 2019, 6:47 AM UTC (cb51eb7)">Diff</a>

### CODE REVIEW

File: spec.bs [hunk 1]
```diff
@@ -315,7 +315,7 @@ To <dfn>register an import map</dfn> given an {{HTMLScriptElement}} |element|:
       1. [=Report a warning to the console=] that an invalid address was given for the specifier key |specifierKey|; since |specifierKey| ended in a slash, so must the address.
       1. [=Continue=].
     1. Set |normalized|[|specifierKey|] to |addressURL|.
-  1. Return the result of [=map/sorting=] |normalized|, with an entry |a| being less than an entry |b| if |a|'s [=map/key=] is [=longer or code unit less than=] |b|'s [=map/key=].
+  1. Return the result of [=map/sorting=] |normalized|, with an entry |a| being less than an entry |b| if |b|'s [=map/key=] is [=code unit less than=] |a|'s [=map/key=].
```
Comment (guybedford) [inline spec.bs]: Are you making the assumption that a code unit outside the end of the string is taken to be zero and thus always less than any other code unit? That seems like something that shouldn't necessarily be an implicit assumption to me.

Comment (hiroshige-g) [reply spec.bs]: https://infra.spec.whatwg.org/#code-unit-less-than explicitly handles prefixes (Steps 1 and 2) and we don't have to care about code unit outside the end of strings.

Comment (guybedford) [reply spec.bs]: Ah perfect, glad to see this is well-defined.

**Comment by marcoscaceres** [comment]
@hiroshige-g please be sure to ping me once you've joined the group and I can click the magic buttons to unblock this. Thanks again!

**Comment by hiroshige-g** [comment]
@marcoscaceres Thanks for handling! It seems I'm now a WICG participant.

**Comment by marcoscaceres** [comment]
Ok, all green now. Good to merge!

**Comment by hiroshige-g** [comment]
> unable to create a test that failed before this change and passed after it.

I think this is fine, because:
In the reference implementation before this PR the sort order was `if the length is greater, or equal and a greater code unit`, not `if the length is greater, or a greater code unit` as specified, and thus the resolution results by the ref impl were already correct.

**Pull request merged**